dm/syncer/data_validator.go

```diff
@@ -366,14 +387,12 @@ func (v *DataValidator) errorProcessRoutine() {
 func (v *DataValidator) waitSyncerSynced(currLoc binlog.Location) error {
 	syncLoc := v.syncer.getFlushedGlobalPoint()
 	cmp := binlog.CompareLocation(currLoc, syncLoc, v.cfg.EnableGTID)
-	switch {
-	case cmp < 0:
-		return nil
-	case cmp == 0:
+	if cmp >= 0 && !v.reachedSyncer.Load() {
 		v.reachedSyncer.Store(true)
+		v.L.Info("validator progress reached syncer")
+	}
+	if cmp <= 0 {
 		return nil
-	default:
-		v.reachedSyncer.Store(true)
```

Comment:
again I forget what's the meaning of `reachedSyncer` 😂 and seems if we lucky always have `currLoc < syncLoc`, `reachedSyncer` will always be false

Reply:
only when `reachedSyncer`=true, validator will start mark rows failed to validate as error rows to reduce chance of false-positive.

>  if we lucky always have currLoc < syncLoc, reachedSyncer will always be false

you're right. we need to make sure `reachedSyncer=true` at some time. will leave a todo and fix in next pr.

Reply:
`reachedSyncer` is not clear for its usage, renamed in later pr.